client/client.go

```diff
@@ -106,6 +106,14 @@ type Info struct {
 	// configauth.ServerAuthenticator implementations tied to the receiver for
 	// this connection.
 	Auth AuthData
+
+	// Metadata is the request metadata from the client connecting to this connector.
+	Metadata *Metadata
```

Comment:
Cannot use pointer here because this allows me to do:

```
*(FromContext(ctx).Metadata) = myMetadata
```

This will change the value in the context.

Reply:
@bogdandrutu correct, fixed